packages/wallet-manager/src/vaults/privatekey-vault.ts

```diff
@@ -19,33 +22,41 @@ export class PrivateKeyVault implements Vault<PkVaultOptions> {
    * one account with the fallowing secret
    */
   constructor(options: PkVaultOptions) {
+    this.provider = options.provider;
     if (options.secret) {
       this.#privateKeys = [options.secret];
     } else {
-      this.#privateKeys = options.accounts || [Wallet.generate().privateKey];
+      this.#privateKeys = options.accounts || [
+        Wallet.generate({
+          provider: options.provider,
+        }).privateKey,
+      ];
     }
   }
 
   serialize(): PkVaultOptions {
     return {
       accounts: this.#privateKeys,
+      provider: this.provider,
```

Comment:
Why do we need to add the provider when serializing a wallet?

Reply:
@Torres-ssf do you mean why are we returning it?

Reply:
Yeah, I'm with @Torres-ssf on this one.

I am trying to understand why we must return the provider in this method.

Reply:
@arboleya @Torres-ssf this is because we need a `provider` to create a `PrivateKeyVault` now. It used to be optional. I actually tried omitting `provider` from the returned values by `serialize`, but it quickly resulted in usages like this one here breaking:

![image](https://github.com/FuelLabs/fuels-ts/assets/39617427/af59179e-92c6-4642-b91f-d302ec4173b1)


What do you guys suggest?

Reply:
Ok, then it's necessary. 👍